### ISSUE TITLE
SFT validation loss of minillm

### ISSUE THREAD

**Issue by chentianyangWHU** [microsoft/LMOps#48]
I did sft (supervised fine-tune) on minillm with the polly dataset by the following order: bash scripts/gpt2/sft/sft_large.sh, and I got the running logs as below:

```
============================== EXP at 2023-07-24 11:19:35 ==============================
dev | avg_loss: 2.50592041015625 | {'exact_match': 0.0, 'rougeL': 7.2973}
......
[2023-07-24 11:53:25] train | epoch   0 | Iter:   1428/ 14290 | global iter:   1428/ 14290 | loss: 1.8361 | ds_loss: 0.0000 | lr: 4.8781e-05 | scale:  2048.0000 | micro time: 1.180 | step time: 1.187
dev | avg_loss: 2.17413330078125 | {'exact_match': 2.6, 'rougeL': 19.8483}
......
[2023-07-24 12:26:31] train | epoch   1 | Iter:   2856/ 14290 | global iter:   2856/ 14290 | loss: 1.5378 | ds_loss: 0.0000 | lr: 4.5241e-05 | scale:  4096.0000 | micro time: 1.203 | step time: 1.194
dev | avg_loss: 2.283782958984375 | {'exact_match': 3.7, 'rougeL': 23.8785}
......
[2023-07-24 12:58:48] train | epoch   2 | Iter:   4284/ 14290 | global iter:   4284/ 14290 | loss: 0.6872 | ds_loss: 0.0000 | lr: 3.9729e-05 | scale:  8192.0000 | micro time: 1.192 | step time: 1.194
dev | avg_loss: 2.521240234375 | {'exact_match': 3.6, 'rougeL': 25.4109}
......
[2023-07-24 13:32:01] train | epoch   3 | Iter:   5716/ 14290 | global iter:   5716/ 14290 | loss: 0.4638 | ds_loss: 0.0000 | lr: 3.2760e-05 | scale:  8192.0000 | micro time: 1.182 | step time: 1.182
dev | avg_loss: 2.7791748046875 | {'exact_match': 3.9, 'rougeL': 26.4599}
......
[2023-07-24 14:04:43] train | epoch   4 | Iter:   7144/ 14290 | global iter:   7144/ 14290 | loss: 0.2177 | ds_loss: 0.0000 | lr: 2.5055e-05 | scale: 16384.0000 | micro time: 1.193 | step time: 1.190
dev | avg_loss: 2.98834228515625 | {'exact_match': 4.1, 'rougeL': 27.4338}
......
[2023-07-24 14:37:37] train | epoch   5 | Iter:   8572/ 14290 | global iter:   8572/ 14290 | loss: 0.1473 | ds_loss: 0.0000 | lr: 1.7361e-05 | scale: 16384.0000 | micro time: 1.183 | step time: 1.183
dev | avg_loss: 3.14642333984375 | {'exact_match': 4.0, 'rougeL': 27.7023}
......
[2023-07-24 15:10:29] train | epoch   6 | Iter:  10000/ 14290 | global iter:  10000/ 14290 | loss: 0.0627 | ds_loss: 0.0000 | lr: 1.0407e-05 | scale: 32768.0000 | micro time: 1.189 | step time: 1.188
dev | avg_loss: 3.27813720703125 | {'exact_match': 4.2, 'rougeL': 28.5247}
......
[2023-07-24 15:43:35] train | epoch   7 | Iter:  11432/ 14290 | global iter:  11432/ 14290 | loss: 0.0307 | ds_loss: 0.0000 | lr: 4.8715e-06 | scale: 32768.0000 | micro time: 1.194 | step time: 1.192
dev | avg_loss: 3.38690185546875 | {'exact_match': 4.0, 'rougeL': 28.6034}
......
[2023-07-24 16:16:19] train | epoch   8 | Iter:  12860/ 14290 | global iter:  12860/ 14290 | loss: 0.0184 | ds_loss: 0.0000 | lr: 1.3279e-06 | scale: 32768.0000 | micro time: 1.197 | step time: 1.192
dev | avg_loss: 3.46453857421875 | {'exact_match': 4.7, 'rougeL': 28.4784}
......
[2023-07-24 16:49:04] train | epoch   9 | Iter:  14288/ 14290 | global iter:  14288/ 14290 | loss: 0.0097 | ds_loss: 0.0000 | lr: 1.0002e-07 | scale: 65536.0000 | micro time: 1.194 | step time: 1.194
dev | avg_loss: 3.4901123046875 | {'exact_match': 4.2, 'rougeL': 29.0872}
```

The experiment records above show that as the training progresses, both the exact_match and rougeL scores increase overall, which is consistent with expectations. However, it is puzzling that the validation set loss (avg_loss) is gradually increasing  as well. According to common knowledge in deep learning, an increase in the loss function contradicts an increase in recognition/detection accuracy. How can we explain this phenomenon?

**Comment by t1101675** [comment]
This is a common phenomenon in natural language generation where validation loss and final metrics do not always correlate. A direct explanation is that the validation loss is only a proxy of the final objective, and the optimization direction does not always match. A detailed explanation can be found in this [link](https://stats.stackexchange.com/questions/282160/how-is-it-possible-that-validation-loss-is-increasing-while-validation-accuracy).

**Comment by chentianyangWHU** [comment]
> This is a common phenomenon in natural language generation where validation loss and final metrics do not always correlate. A direct explanation is that the validation loss is only a proxy of the final objective, and the optimization direction does not always match. A detailed explanation can be found in this [link](https://stats.stackexchange.com/questions/282160/how-is-it-possible-that-validation-loss-is-increasing-while-validation-accuracy).

thanks a lot for answering my question!
